app/_kong_plugins/ai-sanitizer/index.md

```diff
@@ -44,22 +43,55 @@ related_resources:
+
 By integrating with an external PII service, the plugin ensures compliance with data privacy regulations while preserving the usability of request data.
 It supports multiple sanitization modes, including replacing sensitive information with fixed placeholders or generating synthetic replacements that retain category-specific characteristics.
 
-Additionally, AI Sanitizer offers an optional restoration feature, allowing the original data to be reinstated in responses when needed.
+Additionally, AI Sanitizer offers an optional restoration feature, allowing the original request data to be reinstated in responses when needed.
 
 {% include plugins/ai-plugins-note.md %}
 
 The AI Sanitizer plugin uses the AI PII Anonymizer Service, which can run in a Docker container, to detect and sanitize sensitive data. See the [tutorial on configuring the AI Sanitizer plugin](/how-to/protect-sensitive-information-with-ai/) for more information on how to configure the plugin with the AI PII Anonymizer Service.
 
 ## How it works
 
+The AI Sanitizer plugin can be applied to input data (requests), output data (responses) {% new_in 3.12 %}, or both.
+Here's how it works if you apply it to both requests and responses:
+
 1. The plugin intercepts the request body and sends it to the external PII service.
 1. The PII service detects sensitive data and applies the chosen sanitization method (placeholders or synthetic replacements).
 1. The sanitized request is forwarded upstream with the AI Proxy or AI Proxy Advanced plugin.
-1. If restoration is enabled, the plugin restores original data in responses before returning them to the client.
+1. {% new_in 3.12 %} On the way back, the plugin intercepts the response body and sends it to the external PII service.
+1. The PII service detects sensitive data and applies the chosen sanitization method (placeholders or synthetic replacements).
+1. If restoration is enabled, the plugin restores the original request data in responses before returning them to the client.
+
+<!--vale off-->
+{% mermaid %}
+sequenceDiagram
+    autonumber
+    participant Client
+    participant Plugin as AI Sanitizer
+    participant PII as PII Service
+    participant Proxy as AI Proxy/Advanced
+    participant AI as Upstream AI Service
+    
+    Client->>Plugin: Send request
+    Plugin->>PII: Intercept & send request body
+    PII->>PII: Detect sensitive data in request
+    PII->>Plugin: Return sanitized request<br/>(placeholders/synthetic data)
+    Plugin->>Proxy: Forward sanitized request
+    Proxy->>AI: Process sanitized request    
+    AI->>Proxy: Return AI response
+    Proxy->>Plugin: Forward response
+    Plugin->>PII: Intercept & send response body
+    PII->>PII: Detect sensitive data in response
+    PII->>Plugin: Return sanitized response<br/>(placeholders/synthetic data)
+    Plugin->>Client: Return sanitized response<br>(with optional restored request data)
```

Comment:
```suggestion
    Plugin->>Client: Return sanitized response
```

We don't recover redacted PII when we sanitize the response. Otherwise, the redacted data will appear again.